types/index.d.ts

```diff
@@ -1,32 +1,36 @@
 import { FastifyPluginCallback } from 'fastify';
 import { Cluster, Redis, RedisOptions } from 'ioredis';
 
-export interface FastifyRedisNamespacedInstance {
-  [namespace: string]: Redis;
-}
-
-export type FastifyRedis = FastifyRedisNamespacedInstance & Redis;
+type FastifyRedisPlugin = FastifyPluginCallback<fastifyRedis.FastifyRedisPluginOptions>
 
 declare module 'fastify' {
   interface FastifyInstance {
-    redis: FastifyRedis;
+    redis: fastifyRedis.FastifyRedis;
   }
 }
 
-export type FastifyRedisPluginOptions = (RedisOptions &
-{
-  url?: string;
-  namespace?: string;
-}) | {
-  client: Redis | Cluster;
-  namespace?: string;
-  closeClient?: boolean;
-}
+declare namespace fastifyRedis {
+
+  export interface FastifyRedisNamespacedInstance {
+    [namespace: string]: Redis;
+  }
+  
+  export type FastifyRedis = FastifyRedisNamespacedInstance & Redis;
+  
+  export type FastifyRedisPluginOptions = (RedisOptions &
+  {
+    url?: string;
+    namespace?: string;
+  }) | {
+    client: Redis | Cluster;
+    namespace?: string;
+    closeClient?: boolean;
+  }
 
-/**
- * @deprecated Use `FastifyRedisPluginOptions` instead
```

Comment:
i did not keep this deprecated type...
Breaking change?

Reply:
Let keep it for now

Reply:
done